recipes-devtools/ghc/files/0009-Fix-segfault-in-retainer-profiling-when-using-multiple-cores-5909.patch

```diff
@@ -0,0 +1,56 @@
+From a5879a6c2412452fbda8c96e9d921c35279b9d9d Mon Sep 17 00:00:00 2001
+From: Simon Marlow <marlowsd@gmail.com>
+Date: Tue, 19 Feb 2013 09:58:31 +0000
+Subject: [PATCH] Fix segfault in retainer profiling when using multiple cores
+ (#5909)
+
+Thanks to @akio on the ticket for the diagnosis and the patch.  I
+modified the comments a bit.
+---
+ rts/sm/Evac.c | 17 +++++++++++++++--
+  1 file changed, 15 insertions(+), 2 deletions(-)
+
+--- a/rts/sm/Evac.c
++++ b/rts/sm/Evac.c
+@@ -33,7 +33,7 @@ StgWord64 whitehole_spin = 0;
+ #define HEAP_ALLOCED_GC(p) HEAP_ALLOCED(p)
+ #endif
+ 
+-#if !defined(PARALLEL_GC)
++#if !defined(PARALLEL_GC) || defined(PROFILING)
+ #define copy_tag_nolock(p, info, src, size, stp, tag) \
+         copy_tag(p, info, src, size, stp, tag)
+ #endif
+@@ -114,6 +114,17 @@ copy_tag(StgClosure **p, const StgInfoTa
+         const StgInfoTable *new_info;
+         new_info = (const StgInfoTable *)cas((StgPtr)&src->header.info, (W_)info, MK_FORWARDING_PTR(to));
+         if (new_info != info) {
++#ifdef PROFILING
++            // We copied this object at the same time as another
++            // thread.  We'll evacuate the object again and the copy
++            // we just made will be discarded at the next GC, but we
++            // may have copied it after the other thread called
++            // SET_EVACUAEE_FOR_LDV(), which would confuse the LDV
++            // profiler when it encounters this closure in
++            // processHeapClosureForDead.  So we reset the LDVW field
++            // here.
++            LDVW(to) = 0;
++#endif
```

Comment:
IIUC this is the change for the default GHC6 builds in OpenXT. That comment is consistent with the segfault happening intermittently.